pyop2/compilation.py

```diff
@@ -54,6 +55,31 @@ def _check_hashes(x, y, datatype):
     return False
 
 
+@collective
+def single_out_a_rank(comm):
+    rank = MPI.Comm.Get_rank(comm)
+    debug('Using file-based mechanism on %d', rank)
+    fname = os.path.join("/tmp", "fb.%d" % rank)
```

Comment:
Use `mktmp` or whatever it's called to get a guaranteed unique directory name, broadcast that, and then write into the subdirectory.  Otherwise, three things:

1. You might have stuff left around from a previous run
2. /tmp might not be the right place
3. Someone else might be writing to `/tmp/fb.rank`